src/core/dbus-execute.c

```diff
@@ -3868,6 +3888,50 @@ int bus_exec_context_set_transient_property(
 
                 return 1;
 
+        } else if (streq(name, "CgroupNFTSet")) {
+                int family;
+                const char *table, *set, *family_str;
+
+                r = sd_bus_message_read(message, "(iss)", &family, &table, &set);
+                if (r < 0)
+                        return r;
+
+                if (family > 0) {
+                        family_str = nfproto_to_string(family);
+                        if (!family_str)
+                                return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Cgroup NFT family not valid");
+                }
+
+                if (isempty(table))
+                        return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Cgroup NFT table name not valid");
+
+                if (isempty(set))
+                        return sd_bus_error_set(error, SD_BUS_ERROR_INVALID_ARGS, "Cgroup NFT set name not valid");
```

Comment:
some validation should take place here. i.e. we should definitely not allow control characters. I am not sure what naming regime nft enforces, but at the very least string_is_safe() should be used here

Reply:
Found some discussion on [NFT Bugzilla](https://bugzilla.netfilter.org/show_bug.cgi?id=1175).

> Identifiers begin with an alphabetic character (a-z,A-Z), followed zero or more alphanumeric characters (a-z,A-Z,0-9) and the characters slash (/), backslash (\), underscore (_) and dot (.).

> The limit on set identifiers appears to be 31 characters, conflicting with the wiki on this point. It also appears that the limit on chain identifiers is also 31 characters.

Reply:
Added name validation according to the above. It seems at least `-` would have been also OK.